modules/apps/content-dashboard/content-dashboard-web/src/main/java/com/liferay/content/dashboard/web/internal/data/provider/ContentDashboardDataProvider.java

```diff
@@ -27,117 +31,190 @@
 import com.liferay.portal.search.searcher.SearchResponse;
 import com.liferay.portal.search.searcher.Searcher;
 
-import java.util.ArrayList;
-import java.util.Collections;
+import java.util.Collection;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 /**
  * @author David Arques
  */
 public class ContentDashboardDataProvider {
 
 	public ContentDashboardDataProvider(
-		Aggregations aggregations, SearchContext searchContext,
-		Searcher searcher,
+		Aggregations aggregations,
+		ContentDashboardItemFactoryTracker contentDashboardItemFactoryTracker,
+		ContentDashboardSearchContextBuilder
+			contentDashboardSearchContextBuilder,
+		Locale locale, Searcher searcher,
 		SearchRequestBuilderFactory searchRequestBuilderFactory) {
 
 		_aggregations = aggregations;
-		_searchContext = searchContext;
+		_contentDashboardItemFactoryTracker =
+			contentDashboardItemFactoryTracker;
+		_locale = locale;
 		_searcher = searcher;
-		_searchRequestBuilderFactory = searchRequestBuilderFactory;
+
+		_searchRequestBuilder = searchRequestBuilderFactory.builder(
+			contentDashboardSearchContextBuilder.build());
 	}
 
-	public List<AssetCategoryMetric> getAssetCategoryMetrics(
-		String[] audienceAssetCategoryIds, String[] stageAssetCategoryIds) {
+	public Optional<AssetVocabularyMetric> getAssetVocabularyMetric(
```

Comment:
The name should be getAssetVocabularyMetricOptional

Reply:
true!